Automattic/wp-calypso
# Unify Design Preview: Support the font selection for the designs that don't have own style variations

Related to https://github.com/Automattic/wp-calypso/issues/78419

## Proposed Changes

* Introduce a feature flag, `signup/design-picker-preview-fonts`, to control this feature
* Display the font selection for the designs without style variations
* Apply the selected font variation to the preview
* Apply the selected font variation when you continue to activate the theme
* Display the “Unlock custom styles” modal when you select a font variation on your free site

**Follow-up tasks**
* The default font styles cannot be loaded as the Google API might not contain that font.
* Display another copy and modal to unlock the premium styles (colors & fonts)
* Add track events when selecting styles (colors & fonts)
* Support colors

![image](https://github.com/Automattic/wp-calypso/assets/13596067/43dbde8f-f48f-46d8-baf2-43bcf94cdc3d)

https://github.com/Automattic/wp-calypso/assets/13596067/10171d65-9320-4d7b-913c-26fb385223f5

## Testing Instructions



* Go to `/setup?siteSlug=<your_site>`
* Continue until you land on the Design Picker
* Preview the design without the style variation
* On the preview screen,
  * You have to see the font selection
  * The displayed fonts in the preview will be changed when you select any font
  * If the current site is a free site, the “Continue” button will become “Unlock theme” when you select any font.
  * Continue
* When you land on the launchpad, ensure the displayed fonts in the preview are the same as you select

## Pre-merge Checklist
